app/controllers/self_enrollment_controller.rb

```diff
@@ -39,6 +41,13 @@ def enroll_self
 
   private
 
+  def check_early_student_join
+    return unless @course.is_a?(ClassroomProgramCourse) && @course.start.to_date > Time.zone.today
+    return if Alert.exists?(course_id: @course.course_id, type: 'EarlyEnrollmentAlert')
+
+    EarlyEnrollmentAlert.new(course: @course)&.send_email
```

Comment:
I think using `.new` will initialize the object, but will not save it in the database (so we'd still send out a new email each time it happens). This should probably be `.create` instead.

Reply:
I have made the required changes and also fixed a typo error. Thank you for the code review 😃.

Also, I didn’t know why .new was working for me locally 😅, so I’m leaving an explanation below in case it’s needed to clarify why it was working previously 😅.

> I initially thought .new was saving the object to the database, but I didn’t understand why it seemed to work that way. After some research, I realized it was actually because of the .update method.
> 
> Here’s why it worked and why it might not always work:
> 
> If the model doesn’t have strict validations, Rails allows .update to save the object, which makes it look like .new is saving it. However, this isn't how Rails is designed to work, and it might break if validations or other changes are added in the future.

For .new i.e saved by .update later on (Previous Implementation):


![101](https://github.com/user-attachments/assets/43b2cc85-eb13-415a-8cbe-bcf072b2c4b4)


For .create and .update (Current Implementation):

![102](https://github.com/user-attachments/assets/c16d3e6e-e231-45c5-9b06-72f2030fa795)

Reply:
Ah, that makes sense, it was updating (and saving for the first time) when it sends the email, for the email_sent_at timestamp.